service/framework.go

```diff
@@ -253,6 +254,10 @@ func newCRDFramework(config Config) (*framework.Framework, error) {
 		message := messagecontext.NewMessage()
 		ctx = messagecontext.NewContext(ctx, message)
 
+		if config.Viper.GetBool(config.Flag.Service.Guest.Update.Enabled) {
+			updateallowedcontext.SetUpdateAllowed(ctx)
+		}
```

Comment:
Since we wired the contexts already this here is only a minimal change to get full control of the process. I like this.